woocommerce/wc-api-dev
# Order Notes: Add `author` to notes, and `added_by_user` flag to note creation

Fixes #40 

This PR adds an `author` to note responses, listing either `system` or the user display name of the user who created the note. This also adds a different flag, `added_by_user`, to the create call, which if true will set the note author to the currently-logged-in user (the user who authenticated the API request). `added_by_user` was chosen to match the [function `add_order_note`](https://docs.woocommerce.com/wc-apidocs/source-class-WC_Order.html#1430-1474).

This also adds tests for listing notes, getting a single note, creating notes, and deleting notes. For the tests, we also need a helper class (like `WC_Helper_Order`) to create test notes. The tests also check deleting notes, but nothing has changed in the API here.

To test:

- Run `phpunit`
- Check out a few order note responses, `/wc/v3/orders/{orderId}/notes/`,  `/wc/v3/orders/{orderId}/notes/{noteId}`
- Request the schema (by using `OPTIONS`) to see that `author` and `added_by_user` have been added.
- Create a note by POSTing content, see [these docs for how](http://woocommerce.github.io/woocommerce-rest-api-docs/#create-an-order-note), you can also add `added_by_user: true` to set the note author to your authenticated user.

Note: We can hold off release for the composer update, this isn't an urgent issue.
